### PR TITLE
py3: cast dict keys to list

### PR DESCRIPTION
**Pull request by kshtsk**

### CODE REVIEW

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/nuke/__init__.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def nuke_one(ctx, target, should_unlock, synch_clocks, reboot_all,
         ret = target
     else:
         if should_unlock:
-            unlock_one(ctx, target.keys()[0], ctx.owner)
+            unlock_one(ctx, list(target.keys())[0], ctx.owner)
```
Comment (toabctl) [inline teuthology/nuke/__init__.py]: not sure how many elements `target.keys()` has, but if there are a lot, it's faster to really just get the first element with `next(iter(target.keys()))` instead of keeping all keys in memory just to get the first one.
Same comment is true for other places where you do that.
But I'm fine with keeping it as it is if the list is not super large.

File: teuthology/task/kernel.py [hunk 2]
```diff
@@ -682,7 +682,7 @@ def wait_for_reboot(ctx, need_install, timeout, distro=False):
     starttime = time.time()
     while need_install:
         teuthology.reconnect(ctx, timeout)
-        for client in need_install.keys():
+        for client in list(need_install.keys()):
```
Comment (toabctl) [inline teuthology/task/kernel.py]: Is that change needed? Iterating over keys should just work fine in py3, too.

File: tox.ini [hunk 3]
```diff
@@ -52,6 +52,13 @@ deps=
   flake8
 commands=flake8 --select=F,E9 {posargs:teuthology scripts}
 
+[testenv:flake8-py3]
```
Comment (toabctl) [inline tox.ini]: Do you really want that commit in this PR? I think it's from https://github.com/ceph/ceph/pull/32222

File: teuthology/kill.py [hunk 4]
```diff
@@ -239,7 +239,7 @@ def nuke_targets(targets_dict, owner):
         stdout=subprocess.PIPE,
         stderr=subprocess.STDOUT)
     for line in iter(proc.stdout.readline, ''):
-        line = line.replace('\r', '').replace('\n', '')
+        line = line.encode().replace('\r', '').replace('\n', '')
```
Comment (tchaikov) [inline teuthology/kill.py]: might want to put
```py
line = line.encode().replace(b'\r', b'').replace(b'\n', b'')
```

File: teuthology/nuke/__init__.py (hunk 1 above)
Comment (kshtsk) [reply teuthology/nuke/__init__.py]: You're right, and sometime I use next-iter combination, I think it is not critical here since keys are just an objects.

File: teuthology/task/kernel.py (hunk 2 above)
Comment (kshtsk) [reply teuthology/task/kernel.py]: Yes, it is needed very much because of:
```
RuntimeError: dictionary changed size during iteration
```

**Comment by smithfarm** [comment]
Could you squash the last two commits into one?

File: tox.ini (hunk 3 above)
Comment (kshtsk) [reply tox.ini]: Yes, I want it, as early as possible.

**Comment by kshtsk** [comment]
> Could you squash the last two commits into one?

oh, right, I forgot to do this

File: teuthology/kill.py (hunk 4 above)
Comment (kshtsk) [reply teuthology/kill.py]: nice catch... need to double check

File: teuthology/task/kernel.py (hunk 2 above)
Comment (kshtsk) [reply teuthology/task/kernel.py]: btw, there is a shorter way of getting keys as a list from the dict, I would write like `list(need_install)` but this is probably not so intuitive as with exclusively using keys.

File: teuthology/kill.py (hunk 4 above)
Comment (kshtsk) [reply teuthology/kill.py]: I've dropped this commit, and give it a try in another PR.

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy
